src/main/java/com/abdullah/coding/challenge/controllers/AdminController.java

```diff
@@ -0,0 +1,43 @@
+package com.abdullah.coding.challenge.controllers;
+
+import com.abdullah.coding.challenge.Response;
+import com.abdullah.coding.challenge.dto.CabDTO;
+import com.abdullah.coding.challenge.services.AdminService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.PatchMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("admin")
+public class AdminController {
+    @Autowired
+    private AdminService service;
+
+    @PatchMapping("management/cancel-booking")
+    public ResponseEntity<?> cancelBooking(@RequestParam("bookCode") String bookingCode) {
```

Comment:
**Why "bookCode" and not "bookingCode" in the value of the request param?**
- Entirely dependent on the convention a team agrees to follow. I could make it shorter or make it longer, at the end it depends on the agreed upon convention. I have noticed enterprise applications using cryptic query parameter names. So, all in all, entirely depends on the convention a team comes up with.

**Why have we used camel case instead of lower case with dashes?**
- Entirely depends on the convention one agrees to follow. Refer to the above answer.

**Why are we using "PATCH" and not "PUT"?**
- PUT means that you are updating a resource including all of its properties. PATCH means that you are updating one or more properties of a resource. Logically, PATCH made more sense.